Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,53 @@
+lockfail() {
+  odie <<EOS
+Another active Homebrew process is already in process.
+Please wait for it to finish or terminate it to continue.
+EOS
+}
+
+lock() {
+  local name
+  local lock_dir
+  local lock_pid
+  local lock_pid_lock
+  local pid
+
+  name="$1"
+  lock_dir="/tmp/homebrew-${name}.lock"
```

Comment:
Maybe use `TMPDIR` if set instead as OS X differentiates between system and use temporary directories.